src/Symfony/Bundle/SecurityBundle/Command/UserPasswordEncoderCommand.php

```diff
@@ -0,0 +1,152 @@
+        $table
+            ->setHeaders(array('Key', 'Value'))
+            ->addRow(array('Raw password', $input->getOption('show-password') ? $password : str_repeat('*', mb_strlen($password))))
+            ->addRow(array('Salt', $salt))
+            ->addRow(array('Encoded password', $encodedPassword))
+        ;
+
+        $table->render();
+    }
+
+    /**
+     * Create the password question to ask the user for the password to be encoded.
+     *
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     *
+     * @return Question
+     */
+    private function createPasswordQuestion(InputInterface $input, OutputInterface $output)
+    {
+        $passwordQuestion = new Question('<question>Type the password you want to encode please:</question> ');
+
+        $passwordQuestion->setValidator(function ($value) {
+            if ('' === trim($value)) {
+                throw new \Exception('The password must not be empty.');
+            }
+
+            return $value;
+        });
+        $passwordQuestion->setHidden(true);
+        $passwordQuestion->setMaxAttempts(20);
+
+        return $passwordQuestion;
+    }
+
+    /**
+     * Create the question that asks for the salt to perform the encoding.
+     * If there is no provided salt, a random one is automatically generated.
+     *
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     *
+     * @return Question
+     */
+    private function createSaltQuestion(InputInterface $input, OutputInterface $output)
+    {
+        $saltQuestion = new Question('<question>Provide a salt (if none, a salt will be generated):</question> ');
+
+        $that = $this;
+        $saltQuestion->setValidator(function ($value) use ($output, $that) {
+            if ('' === trim($value)) {
+                $value = hash('sha512', $that->getContainer()->get('security.secure_random')->nextBytes(30));
+
+                $output->writeln("\n<comment>The salt has been generated: </comment>".$value);
+                $output->writeln(sprintf("<comment>Make sure that your salt storage field fits this salt length: %s chars.</comment>\n", strlen($value)));
+            }
+
+            return $value;
+        });
+
+        return $saltQuestion;
+    }
+
+    /**
+     * Create the question that asks for the configured user class.
+     *
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     *
+     * @return Question
+     */
+    private function createUserClassQuestion(InputInterface $input, OutputInterface $output)
+    {
+        $userClassQuestion = new Question('<question>Provide your configured user class:</question> ');
+        $userClassQuestion->setAutocompleterValues(array('Symfony\Component\Security\Core\User\User'));
+
+        $userClassQuestion->setValidator(function ($value) use ($output) {
+            if ('' === trim($value)) {
+                $value = 'Symfony\Component\Security\Core\User\User';
```

Comment:
@saro0h Would it be possible to load the configuration and propose a choice amongst the configure user classes? That would be much more easier for end users (typing a fully-qualified class name without a typo is hard on the CLI -- at least for me.)